Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -432,6 +452,26 @@ private void OnShuttleStartup(EntityUid uid, ArrivalsShuttleComponent component,
         EnsureComp<PreventPilotComponent>(uid);
     }
 
+    // Moffstation - Start - First arrivals event
+    private void OnFirstArrival(EntityUid uid, ArrivalsShuttleComponent component, FirstArrivalEvent ev)
+    {
+        // Fixes to problems exclusive to the group arrivals start
+        if (_cfgManager.GetCVar(CCVars.StartAtArrivals))
+        {
+            if (_station.GetStationInMap(Transform(uid).MapID) is not { } station)
+                return;
+            // Refills all the station's batteries, gives engi more leeway since they have to wait to arrive at the station
+            var grid = _station.GetLargestGrid(Comp<StationDataComponent>(station));
```

Comment:
I think this should not be specific to the largest grid, but apply to all grids composing the station, in case the station is in fact multi grid somehow.